glmmbernoulli.qmd

```diff
@@ -0,0 +1,523 @@
+This confirms our earlier impression that the main distinction is between women with children and those without and, for those who do have children, the number of children is not an important distinction.
+
+After incorporating a new variable `ch` --- an indicator of whether the woman
+has any children --- in the data,
+
+```{julia}
+contrasts[:ch] = HelmertCoding();
+contra = Table(contra; ch=contra.livch .!= "0")
+```
+
+```{julia}
+#| code-fold: true
+let
+  df = DataFrame(contra)
+  describe(df, :mean, :min, :median, :max, :nunique, :eltype)
+end
+```
+
+we fit a reduced model.
+
+```{julia}
+com02 = let
+  form = @formula(use ~ 1 + ch + age * age * urban + (1 | dist))
+  fit(MixedModel, form, contra, Bernoulli(); contrasts, nAGQ)
+end
+```
+
+Comparing this model to the previous model
+
+```{julia}
+MixedModels.likelihoodratiotest(com02, com01)
+```
+
+indicates that the reduced model is adequate.
+
+Apparently neither the second-order interaction `age & urban` nor the third-order interaction `age & age & urban` is significant and we fit a model without these terms.
+
+```{julia}
+com03 = let
+  form = @formula(use ~ 1 + urban + ch + age * age + (1 | dist))
+  fit(MixedModel, form, contra, Bernoulli(); contrasts, nAGQ)
+end
+```
+
+A likelihood ratio test
+
+```{julia}
+MixedModels.likelihoodratiotest(com03, com02)
+```
+
+indicates that these terms can safely be eliminated.
+
+A plot of the smoothed observed proportions versus centered age by `urban` and `ch`, @fig-contradata2,
+
+```{julia}
+#| code-fold: true
+#| fig-cap: Smoothed relative frequency of contraception use versus centered age for women in the 1989 Bangladesh Fertility Survey.  The livch factor has been collapsed to children/nochildren.
+#| label: fig-contradata2
+draw(
+  data(contra) *
+  mapping(
+    :age => "Centered age (yr)",
+    :use => ==("Y") => "Frequency of contraceptive use";
+    col=:urban => renamer(["N" => "Rural", "Y" => "Urban"]),
+    color=:ch => "Children",
+  ) *
+  smooth();
+  figure=(; resolution=(800, 450)),
+)
+```
+
+indicates that all four groups have a quadratic trend with respect to age but the location of the peak proportion is shifted for those without children relative to those with children.
+Incorporating an interaction of `age` and `ch` allows for such a shift.
+
+```{julia}
+com04 = let
+  form =
+    @formula(use ~ 1 + urban + ch * age + age & age + (1 | dist))
+  fit(MixedModel, form, contra, Bernoulli(); contrasts, nAGQ)
+end
+```
+
+Comparing this fitted model to the previous one
+
+```{julia}
+MixedModels.likelihoodratiotest(com03, com04)
+```
+
+confirms the usefulness of this term.
+
+A series of such model fits led to a model with random effects for the combinations of `dist` and `urban`, because differences between urban and rural women in the same district were comparable to differences between districts, even after accounting for an effect of `urban` at the fixed-effects (or population) level.
```

Comment:
I think this is a really interesting use of random effects from a casual user perspective and it might be worthwhile to emphasize why it's valid. A rule of thumb for less expert users is that the same variables should not appear as fixed effects and as blocking variables, and this seems to go against that. The magic is that "the urban and rural parts of the same district tend to be so different that there are effectively two districts" and "we can nonetheless separate out the effect of `urban` from district-to-district variability".